app/controllers/blogs_controller.rb

```diff
@@ -0,0 +1,39 @@
+class BlogsController < ApplicationController
+    # Added before_action to restrict app access until after successful login
+    before_action :require_user, only: [:index, :show]
```

Comment:
Filters like `:require_user` should be applied before anything that should be restricted, not just things that render a view.

Without it, it's possible to create a blog without being logged in.

Reply:
Could you elaborate a bit more on this? 

I added the ```:require_user``` to ensure a user should be logged in before creating a blog..

Reply:
@SughoshRao at the moment, you're only ensuring that the user is logged in for the `:index` and `:show` actions. There's nothing to stop someone from sending a POST request (e.g. via cURL) and creating a blog (including assigning it to a different user).

So it really should be something like:
```ruby
before_action :require_user, except: [:some_action_that_should_be_available_to_the_public]
```
(You can set the list to _include_ using `only` or to _exclude_ using `except`. If you don't specify either of these, it will be applied to all actions by default, which is likely the case you would want here.)

Does that make more sense?

Reply:
Oh I see. Yeah this helps, thanks!